kiwi/xml_state.py

```diff
@@ -1808,6 +1809,20 @@ def set_derived_from_image_uri(self, uri):
             ''')
             log.warning(message.format(uri))
 
+    def set_root_partition_uuid(self, uuid):
```

Comment:
I have to admit I am surprised you use xml_state instance to share the part uuid between DiskBuilder instance and the BootImage instance :thinking: I have no strong opinion against it but I find it unexpected, I am not sure the root partition uuid belongs there. I would expect that most, if no all, the data contained in the xml_state instance is somehow a result of XML file together a kiwi command like.

Wouldn't it make sense having a root_partuuid new attribute in BootImage?

Reply:
Well I was also thinking what would be a good namespace to store that information. The xml_state object transports stateful information about the image description. The state in this namespace covers information that is unique to the build, for example the selected profile. I thought it would not violate the scope of the instance if stateful information that does not directly come from the image description is stored there too. This has the advantage that the information is available across all major steps of the image build because the xml_state object is handed over to any major class of the build process.

The BootImage class is a reader of the information because it writes the .profile file.
The contents of the .profile file are created exclusively from the state of xml_state.
That's why I put the root_partuuid there too. kiwi itself does not need this information,
also the BootImage implementation does not need this information. Adding it to
BootImage felt odd to me as it is not needed.

I can understand the confusion you see when reading the BootImage changes.
I think I will change that code to make clear BootImage doesn't need root_partuuid

Reply:
Added a commit which hopefully makes it more clear ?

Reply:
> I thought it would not violate the scope of the instance if stateful information that does not directly come from the image description is stored there too. This has the advantage that the information is available across all major steps of the image build because the xml_state object is handed over to any major class of the build process.

Right I buy it.

> Added a commit which hopefully makes it more clear ?

Yes thanks, I believe that with the current implementation I would have not been surprised or confused.